src/llvm-gcroot.cpp

```diff
@@ -37,6 +42,74 @@ static struct {
 } jl_gc_frame_stats = {0};
 #endif
 
+typedef std::pair<CallInst*, unsigned> frame_register;
+struct liveness {
+    typedef unsigned id;
+    enum {
+        // an assignment to a gcroot exists in the basic-block
+        // (potentially no live-in from the predecessor basic-blocks)
```

Comment:
could that be explained in a handful more words? having confusing jargon in the comments sort of defeats the explanatory purpose of them